### PR TITLE
Support for new IAM API Key authentication for Assistant.

### PR DESCRIPTION
**Pull request by tanmayb123**
I haven't been able to test this code properly and won't be able to for the next few days. If possible, could someone else please run this code and test it out for me? If not, I'll do it ASAP.

### CODE REVIEW

File: CognitiveConcierge-iOS/CognitiveConcierge/CognitiveConcierge.plist [hunk 1]
```diff
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+        <key>ConversationIamApiKey</key>
+        <string>your_key_here</string>
 	<key>ConversationPassword</key>
 	<string>Vzah3JxwdpuI</string>
```
Comment (stevemar) [inline CognitiveConcierge-iOS/CognitiveConcierge/CognitiveConcierge.plist]: blerg. i didn't realize someone checked in values here... we should remove them all.

File: CognitiveConcierge-iOS/CognitiveConcierge/EntertainmentOptionsViewController.swift [hunk 2]
```diff
@@ -62,8 +62,12 @@ class EntertainmentOptionsViewController: UIViewController {
         convoContext = nil
         
         // Instantiating to start first conversation with Watson.
-        convoService = Conversation(username: configuration?["ConversationUsername"] as! String, password: configuration?["ConversationPassword"] as! String, version: configuration?["ConversationVersion"] as! String)
-        
+        if let iamApiKey = configuration?["ConversationApiKey"] as? String, !iamApiKey.isEmpty {
```
Comment (stevemar) [inline CognitiveConcierge-iOS/CognitiveConcierge/EntertainmentOptionsViewController.swift:65]: LGTM

**Review by stevemar — APPROVED** [review]

File: CognitiveConcierge-iOS/CognitiveConcierge/EntertainmentOptionsViewController.swift (hunk 2 above)
Comment (tanmayb123) [reply CognitiveConcierge-iOS/CognitiveConcierge/EntertainmentOptionsViewController.swift:65]: Small problem: here I refer to "ConversationApiKey" but in the plist it's "ConversationIamApiKey". I'll change it to "ConversationApiKey" in the plist.

File: CognitiveConcierge-iOS/CognitiveConcierge/CognitiveConcierge.plist (hunk 1 above)
Comment (tanmayb123) [reply CognitiveConcierge-iOS/CognitiveConcierge/CognitiveConcierge.plist]: Alright, I'll do that.

**Comment by tanmayb123** [comment]
Travis CI says that I need to rename ConversationV1 to AssistantV1 in all the files. I'll do that by tomorrow and commit.

**Comment by tanmayb123** [comment]
I've run this code on Xcode 9.4, and it all works for me. Travis is complaining about "AssistantV1" not existing because it's using an older version.

**Review by stevemar — APPROVED** [review]

**Pull request merged**